src/parser.rs

```diff
@@ -585,36 +580,54 @@ fn parse_variable_block(
     //Consume the type keyword
     lexer.advance();
     let variables = parse_any_in_region(lexer, vec![KeywordEndVar], |lexer| {
-        let mut variables = vec![];
-        while lexer.token == Identifier {
-            if let Some(variable) = parse_variable(lexer) {
-                variables.push(variable);
-            }
-        }
-        variables
+        parse_variable_list(lexer)
     });
     VariableBlock {
         variables,
         variable_block_type,
     }
 }
 
-fn parse_variable(lexer: &mut ParseSession) -> Option<Variable> {
-    let variable_location = lexer.location();
-    let name = lexer.slice_and_advance();
+fn parse_variable_list(lexer: &mut ParseSession) -> Vec<Variable> {
+    let mut variables = vec![];
+    while lexer.token == Identifier {
+        let mut line_vars = parse_variable_line(lexer);
+        variables.append(&mut line_vars);
+    }
+    variables
+}
+
+fn parse_variable_line(lexer: &mut ParseSession) -> Vec<Variable> {
+    // read in a comma separated list of variable names
+    let mut var_names: Vec<(String, SourceRange)> = vec![];
+    while lexer.token == Identifier {
```

Comment:
It might be a better recovery strategy to loop until we see the Colon instead.
The idea would be : 
Parse Identifier and push to list
Loop while token not colon (or semicolon?)
Allow comma, complain if missing and the current token is the identifier
Push identifier
In the current solution we will only recover on the semicolon.
But my proposal risks an infinite loop. 
Maybe we find something in between? Like don't break on the !allow for the comma, but log an error if last token was not a comma?

Reply:
Expressions in the form of "while not keyword" will indeed cause an infinite loop in some situations. I also encountered a similar problem when I first implemented this. Not breaking on !allow sounds reasonable to me, especially since one is more likely to forget a comma than to forget an identifier.